src/communication/communicator.hpp

```diff
@@ -56,11 +56,11 @@ class communicator {
         // For caching information about each cell
         struct gid_info {
             using connection_list = decltype(std::declval<recipe>().connections_on(0));
-            cell_gid_type gid;
-            cell_gid_type local_group;
-            connection_list conns;
-            gid_info(cell_gid_type g, cell_gid_type lg, connection_list c):
-                gid(g), local_group(lg), conns(std::move(c)) {}
+            cell_gid_type gid;       // global identifier of cell
+            cell_gid_type domain_id; // index of cell in this domain
```

Comment:
domain_id is the id inside the domain, not the id of the domain?

Reply:
Yes. Not the best name.

Reply:
Changed to `index_on_domain`. Verbose, but clearer.